scipy/stats/_stats_py.py

```diff
@@ -100,13 +100,20 @@ def _contains_nan(a, nan_policy='propagate'):
         try:
             contains_nan = np.any(np.isnan(a))
         except TypeError:
-            # Don't know what to do. Fall back to omitting nan values and
-            # issue a warning.
-            contains_nan = False
-            nan_policy = 'omit'
-            warnings.warn("The input array could not be properly "
-                          "checked for nan values. nan values "
-                          "will be ignored.", RuntimeWarning)
+            try:
+                # This can happen when attempting to check nan with np.isnan
+                # for string array (e.g. as in the function `rankdata`).
+                a = np.array(a)
+                nan_str = np.array([np.nan]).astype(str)[0]
```

Comment:
```suggestion
                nan_str = "nan"
```
This would work, at least on my machine. If we think there is some reason that we need the added robustness of `np.array([np.nan]).astype(str)[0]`, there should be a comment, like:

```
# `nan_str = 'nan'` would work, but this is robust to differences in conversion of np.nan to a string
```

Reply:
The reason for this code is that I don't have enough confidence always nan_str is "nan". But let me try the CIs